### ISSUE TITLE
[humble] Can't find pylon/PylonIncludes.h

### ISSUE THREAD

**Issue by bertaveira** [basler/pylon-ros-camera#160]
I am getting an error trying to build the `humble` branch. I have installed the pylon software suite and even tested with the pylon viewer.

However when I try to build I get this error:
`pylon_ros2_camera_component/src/tools/ip_auto_config.cpp:29:10: fatal error: pylon/PylonIncludes.h: No such file or directory`

Any ideas what I am doing wrong?

I installed with the Debian package downloaded from the Basler website. Version 7.3 on both ARM and x86 ubuntu 22.04

**Comment by bertaveira** [comment]
Yeah I had to move to 7.x because the 6.x has no ARM Debian package. I find it weird how in the `humble` branch README it says that we should use 7.2 or later

**Comment by whoutman** [comment]
Unfortunately I bumped into this issue as well.

It seems that the cmake/FindPylon.cmake is not properly included in CMakeLists.txt, as I was able to compile when commenting out the if-statement around the include of this cmake-file, see https://github.com/basler/pylon-ros-camera/blob/humble/pylon_ros2_camera_component/CMakeLists.txt#L39

With this if-statement, no output was generated for the messages in the lines below (I commented those out, see https://github.com/basler/pylon-ros-camera/blob/humble/pylon_ros2_camera_component/CMakeLists.txt#L41-L43), while the following output was reported without the if-statement:
```
->    Pylon include directories: /opt/pylon/include
->    Pylon library directories: /opt/pylon/lib
->    Pylon libraries: pylonbase;pylonutility;GenApi_gcc_v3_1_Basler_pylon;GCBase_gcc_v3_1_Basler_pylon
```

Did I forget to run a specific script? Running e.g. 
`source /opt/pylon/bin/pylon-setup-env.sh /opt/pylon`
does not seem to do the job.

As a reference:
pylon version 7.3.0 (downloaded deb package installer) with a x86 (64 Bit) architecture

**Comment by alblinks** [comment]
I am experiencing the same error. Downgrading to pylon 6.3 worked.

**Comment by Alex-Beh** [comment]
Facing the same issue and the solution by @whoutman works for me. Somehow the cmake cannot find the library but the Pylon_FOUND is set to 1. So the if case for the FindPylon.cmake has been skipped.

I have modified the if case to the following also and managed to compiled the packages with pylon 7.3 [👍](https://emojipedia.org/thumbs-up/)
```
if (NOT ${Pylon_FOUND} OR "$ENV{Pylon_INCLUDE_DIRS}" STREQUAL "")
    include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindPylon.cmake")
endif()
```

**Comment by Ekanshh** [comment]
> Facing the same issue and the solution by @whoutman works for me. Somehow the cmake cannot find the library but the `Pylon_FOUND` is set to 1. So the if case for the FindPylon.cmake has been skipped.
> 
> I have modified the if case to the following also and managed to compiled the packages with pylon 7.3 [+1](https://emojipedia.org/thumbs-up/)
> 
> ```
> if (NOT ${Pylon_FOUND} OR "$ENV{Pylon_INCLUDE_DIRS}" STREQUAL "")
>     include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindPylon.cmake")
> endif()
> ```

Worked for me. :)

**Comment by argolomb** [comment]
> Facing the same issue and the solution by @whoutman works for me. Somehow the cmake cannot find the library but the `Pylon_FOUND` is set to 1. So the if case for the FindPylon.cmake has been skipped.
> 
> I have modified the if case to the following also and managed to compiled the packages with pylon 7.3 [+1](https://emojipedia.org/thumbs-up/)
> 
> ```
> if (NOT ${Pylon_FOUND} OR "$ENV{Pylon_INCLUDE_DIRS}" STREQUAL "")
>     include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindPylon.cmake")
> endif()
> ```

Work for me too.

**Comment by nachovizzo** [comment]
@FrancoisPicardDTI, any plan on fixing this? It would be ideal to be able to build the ROS 2 wrapper without having to patch the build system.

On the README.md it says that Pylon SDK 7.2+ should be installed but this conflicts with the way of finding the package. The provided solution by @Alex-Beh works so seems to be easy to include in the ros2 branch.

What do you think? I'm happy to open a PR to fix this

**Comment by ndahn** [comment]
Commenting out the if statement also worked on ROS noetic!

**Comment by FrancoisPicardDTI** [comment]
Hello
I integrated the fixes suggested by @nachovizzo, PR #179.
Does it solve the issue?
Regards,
Francois
